test/Driver/static-archive.swift

```diff
@@ -0,0 +1,36 @@
+// RUN: %swiftc_driver -driver-print-jobs -target x86_64-apple-macosx10.9 -emit-library %s -module-name ARCHIVER -static 2>&1 > %t.macos.txt
+
+// CHECK: swift
+// CHECK: -o [[OBJECTFILE:.*]]
+
+// CHECK-NEXT: {{(bin/)?}}libtool{{"? }} -static
+// CHECK-DAG: [[OBJECTFILE]]
+// CHECK: -o {{[^ ]+}}
+
+// RUN: %swiftc_driver -driver-print-jobs -target x86_64-unknown-linux-gnu -emit-library %s -module-name ARCHIVER -static 2>&1 > %t.linux.txt
+
+// CHECK: swift
+// CHECK: -o [[OBJECTFILE:.*]]
+
+// CHECK-NEXT: {{(bin/)?}}{{(llvm-)?}}ar{{"? }} crs
+// CHECK-NEXT: {{[^ ]+}}
+
+// RUN: %swiftc_driver -driver-print-jobs -target x86_64-unknown-windows-msvc -emit-library %s -module-name ARCHIVER -static 2>&1 > %t.windows.txt
+
+// CHECK: swift
+// CHECK: -o [[OBJECTFILE:.*]]
+
+// CHECK-NEXT: link
+// CHECK-DAG: [[OBJECTFILE]]
+// CHECK: /OUT:{{[^ ]+}}
+
+// RUN: %swiftc_driver -driver-print-jobs -target x86_64-apple-macosx10.9 -emit-library %s -module-name ARCHIVER -static | %FileCheck -check-prefix INFERRED_NAME_DARWIN %s
+// RUN: %swiftc_driver -driver-print-jobs -target x86_64-unknown-linux-gnu -emit-library %s -module-name ARCHIVER -static | %FileCheck -check-prefix INFERRED_NAME_LINUX %s
+// RUN: %swiftc_driver -driver-print-jobs -target x86_64-unknown-windows-msvc -emit-library %s -module-name ARCHIVER -static | %FileCheck -check-prefix INFERRED_NAME_WINDOWS %s
+
+// INFERRED_NAME_DARWIN: bin{{/|\\\\}}swift{{c?(\.EXE)?}}
+// INFERRED_NAME_DARWIN: -module-name ARCHIVER
+// INFERRED_NAME_DARWIN: libtool -static
+// INFERRED_NAME_DARWIN:  -o libARCHIVER.a
+// INFERRED_NAME_LINUX:   libARCHIVER.a
+// INFERRED_NAME_WINDOWS: ARCHIVER.lib
```

Comment:
Please add a test case for the invalid combination (`-emit-executable -static`) as well as tests for the fallback path using mocked toolchains.

Reply:
Could you clarify what you mean by "tests for the fallback path using mocked toolchains", or point to existing similar tests? I assume you mean ensure that we use the correct tools for e.g. `lld-link`, but I can't see where that's actually tested for the dynamic linker path – there's a line referencing `config.link` in `lit.cfg` but I'm not sure how that feeds into the tests.